daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1602,12 +1604,52 @@ private[lf] object SBuiltin {
       args.get(0) match {
         case SAnyException(ty, innerValue) =>
           machine.ctrl = SEApp(exceptionMessage(ty), Array(SEValue(innerValue)))
+        case SBuiltinException(ArithmeticError) =>
+          // TODO https://github.com/digital-asset/daml/issues/8020
+          //   Return more useful exception message (e.g. include the function and arguments)
+          machine.returnValue = SText("ArithmeticError")
+        case SBuiltinException(ContractError) =>
+          // TODO https://github.com/digital-asset/daml/issues/8020
+          //   Return more useful exception message (e.g. include the template name)
+          machine.returnValue = SText("ContractError")
         case v =>
           crash(s"AnyExceptionMessage applied to non-AnyException: $v")
       }
     }
   }
 
+  /** $any-exception-is-arithmetic-error :: AnyException -> Bool */
+  final case object SBAnyExceptionIsArithmeticError extends SBuiltinPure(1) {
+    override private[speedy] final def executePure(args: util.ArrayList[SValue]): SValue = {
+      args.get(0) match {
+        case SBuiltinException(ArithmeticError) =>
```

Comment:
Have you considered adding a `getSException` similar to how we have `getSInt64`?

Reply:
Nice idea, thanks :)